src/Decoder/DecoderResult.php

```diff
@@ -37,6 +37,12 @@ final class DecoderResult{
 	private string      $data = '';
 	private int         $structuredAppendParity = -1;
 	private int         $structuredAppendSequence = -1;
+	private int         $topLeftX;
+	private int         $topLeftY;
+	private int         $topRightX;
+	private int         $topRightY;
+	private int         $bottomLeftX;
+	private int         $bottomLeftY;
```

Comment:
I'd just add 3 properties with the `FinderPattern` type here. I'm not sure yet what is returned when a pattern is not found - when in doubt, make the fields nullable.